### DISCUSSION TITLE
IMetadataReader now has `GetObjectID()`. How to implement?

### DISCUSSION THREAD

**Discussion by timbze** [linq2db/linq2db#3978] [General]
In 5.0.0 RC2, the method `GetObjectID()` must now be implemented. How do I implement it?

I have a MetadataReader for column and table names to read them as snake_case.

**Comment by MaceWindu** [comment]
It is some kind of hash, used for mapping schema caching. In your case:
- if reader consumes data from attributes - just return constant value, because metadata cannot change in runtime, e.g. check how it is done by [AttributeReader](https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Metadata/AttributeReader.cs#L32)
- if data, returned by your `IMetadataReader` is instance-specific - you need to calculate some unique cache based on content (or you can use `GetHashCode` result if you create just couple of instances)

**Reply by timbze** [reply]
That makes sense. Then I should be fine with the first approach like AttributeReader. Thank you!
